internal/test/builder/builders.go

```diff
@@ -598,6 +615,9 @@ func (m *MachineDeploymentClassBuilder) Build() *clusterv1.MachineDeploymentClas
 	if m.strategy != nil {
 		obj.Strategy = m.strategy
 	}
+	if m.namingStrategy != nil {
```

Comment:
nit: up to you. Technically we can simplify a lot of this by getting rid of all the nil checking. Setting nil on a field that was never set before is a no-op. 

If you want to do it consistently across this file let's do it in a separate PR. But also fine to just keep it as is for me, just noticed it